xwiki-platform-core/xwiki-platform-ckeditor/xwiki-platform-ckeditor-plugins/src/main/resources/xwiki-focusedplaceholder/plugin.js

```diff
@@ -0,0 +1,268 @@
+/*
+ * See the NOTICE file distributed with this work for additional
+ * information regarding copyright ownership.
+ *
+ * This is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU Lesser General Public License as
+ * published by the Free Software Foundation; either version 2.1 of
+ * the License, or (at your option) any later version.
+ *
+ * This software is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this software; if not, write to the Free
+ * Software Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA
+ * 02110-1301 USA, or see the FSF site: http://www.fsf.org.
+ */
+(function() {
+    'use strict';
+
+    var ATTRIBUTE_NAME = 'data-xwiki-focusedplaceholder';
+
+    CKEDITOR.plugins.add('xwiki-focusedplaceholder', {
+
+        /**
+         * Styles that would be applied to the editor by the placeholder text when visible.
+         *
+         * @property {String}
+         */
+        placeholderStyle: '[' + ATTRIBUTE_NAME + ']::before {' +
+            'position: absolute;' +
+            'opacity: .8;' +
+            'color: #aaa;' +
+            'content: attr( ' + ATTRIBUTE_NAME + ' );' +
+            '}' +
+            '.cke_wysiwyg_div[' + ATTRIBUTE_NAME + ']::before {' +
+            'margin-top: 1em;' +
+            '}',
```

Comment:
I'm wondering if it's not better to put the styles in https://github.com/xwiki/xwiki-platform/blob/master/xwiki-platform-core/xwiki-platform-ckeditor/xwiki-platform-ckeditor-ui/src/main/resources/CKEditor/ContentSheet.xml#L202 (which holds custom styles that apply to the edited content). They wont be packaged with the plugin code but at least we don't mix JavaScript and CSS. WDYT?

Reply:
I agree, the style is hard to read in Javascript context, I moved it in the ContentSheet.
Thank you.